bpf/lib/conntrack.h

```diff
@@ -527,14 +528,17 @@ static __always_inline int ipv4_ct_extract_l4_ports(struct __ctx_buff *ctx,
 	 * after data has been invalidated (see handle_ipv4_from_lxc())
 	 */
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
-		return DROP_INVALID;
+		return DROP_CT_INVALID_HDR;
 
 	if (unlikely(ipv4_is_fragment(ip4)))
-		return ipv4_handle_fragment(ctx, ip4, off,
+		return ipv4_handle_fragment(ctx, ip4, off, ct_dir,
 					    (struct ipv4_frag_l4ports *)&tuple->dport);
 #endif
 	/* load sport + dport into tuple */
-	return ctx_load_bytes(ctx, off, &tuple->dport, 4);
+	if (ctx_load_bytes(ctx, off, &tuple->dport, 4) < 0)
+		return DROP_CT_INVALID_HDR;
```

Comment:
Can `ctx_load_bytes` fail if we already checked the packet size earlier (`revalidate_data` in case `ENABLE_IPV4_FRAGMENTS` is defined)? If it cannot fail, maybe we could unconditionally return `CTX_ACT_OK` to reduce complexity?

Alternatively, we could also always perform the `revalidate_data` call and replace the `ctx_load_bytes` call by a direct packet read.